walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,76 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> timeRanges = new ArrayList<TimeRange>();
+        List<TimeRange> attendeeRelevantTimeRanges = new ArrayList<TimeRange>();
+        Collection<String> requestAttendees = request.getAttendees(); 
+        Collection<String> eventAttendees;
+
+        // Add the TimeRanges of events into a List that are relevant to the request's attendees 
+        for(Event event : events){
+	        eventAttendees = event.getAttendees(); 
+	        if(!Collections.disjoint(requestAttendees, eventAttendees)){
+	            attendeeRelevantTimeRanges.add(event.getWhen());
+            }
+        }
+        
+        // Sort the List so that TimeRanges are ordered by when they start
+        Collections.sort(attendeeRelevantTimeRanges, TimeRange.ORDER_BY_START);
+
+        /** 
+         * Check edge cases: Check if duration of request is valid and check if there are 
+         * any relevant time ranges that affect the request.
+         */
+        if(attendeeRelevantTimeRanges.size() == 0){
+            if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+                return timeRanges;
+            }
+            timeRanges.add(TimeRange.WHOLE_DAY);
+            return timeRanges;
+        }
+
+        // Add a TimeRange with a range from the beginning of the day until start of first event if duration is valid
+        TimeRange current = attendeeRelevantTimeRanges.get(0);
+        TimeRange toAdd = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, current.start(), false);
+        if(toAdd.duration() >= request.getDuration()){
+                timeRanges.add(toAdd);
```

Comment:
Nit: Indentation

Reply:
It seems like there are several spaces on github but on my end, there is only a tab. Unsure how to fix so I left it as a tab on my end.